### DISCUSSION TITLE
How to align lualine with nvim-tree when tree open?

### DISCUSSION THREAD

**Discussion by Raagh** [nvim-lualine/lualine.nvim#892] [Configuration]
Hi. So I am trying to align the start of lualine to the end of the nvim-tree when the tree is open. I had a similar issue with barbar.nvim but they provided a solution in their documentation. You can check it [here](https://github.com/romgrk/barbar.nvim#integration-with-filetree-plugins). Is there something similar I can do for lualine?

Barbar example:
![imagen](https://user-images.githubusercontent.com/8405459/202442848-729e6370-6090-4bdd-b069-ef6e073cb95f.png)

Lualine current behavior:
![imagen](https://user-images.githubusercontent.com/8405459/202442763-adde22bf-b6c8-4a2c-9f60-570f70fb03f3.png)

**Comment by fitrh** [comment]
What do you expect lualine to look like?
Is the global status line not suitable for your needs?

**Reply by Raagh** [reply]
I expect the "box with blue background with the text normal" to start where nvim-tree ends just like the tabs from barbar do (in the picture you can see is aligned a few pixels to the right). I will look into global status line I am not familiar with its configuration.

**Comment by shadmansaleh** [comment]
look at the discussion in #671

**Reply by Raagh** [reply]
this discussion seems to be on how to shift it right no? I wan't to shift it left

**Reply by Raagh** [reply]
okay solved it with your discussion, although it feels kind of hacky. I had to enable global status line so it takes the entire screen, then fix it so it aligns perfectly with the end of nvim-tree...

```
local status_ok, lualine = pcall(require, "lualine")
if not status_ok then
  return
end

local nvim_tree_shift =  {
  function ()
    return string.rep(' ',
                      vim.api.nvim_win_get_width(require'nvim-tree.view'.get_winnr()) - 2)
  end,
  cond = require('nvim-tree.view').is_visible,
  color = 'NvimTreeNormal'
}

lualine.setup({
  extensions = { 'nvim-tree', 'toggleterm', 'nvim-dap-ui' },
  theme = 'catppuccin',
  options = {
    globalstatus = true,
  },
  sections = {
     lualine_a = { nvim_tree_shift, "mode" }
  },
})
```

**Reply by Raagh** [reply]
![imagen](https://user-images.githubusercontent.com/8405459/202527955-f4a0b06d-602c-4f72-b5e1-13cade75c237.png)

**Reply by Raagh** [reply]
unfortunately the experience is not great, when toy close nvim-tree it takes like a second to refresh and you have the line just hanging there in the middle of the screen

**Reply by fitrh** [reply]
> unfortunately the experience is not great, when toy close nvim-tree it takes like a second to refresh and you have the line just hanging there in the middle of the screen

Seems you need to reduce `options.refresh.statusline`, the default is 1000ms, you can try 100ms
